types/react-bootstrap/lib/ToggleButtonGroup.d.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+
+declare class ToggleButtonGroup extends React.Component<ToggleButtonGroupProps> { }
+declare namespace ToggleButtonGroup { }
+export = ToggleButtonGroup
+
+interface ToggleButtonGroupProps extends React.HTMLProps<ToggleButtonGroup> {
+    /** Required if `type` is set to "radio" */
+    name?: string;
+    type: "radio" | "checkbox";
+    /**
+     * You'll usually want to use string|number|string[]|number[] here,
+     * but you can technically use any|any[].
+     */
+    defaultValue?: any|any[];
```

Comment:
Please see [Common Mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) (last point). You'll want to reduce this to just `any`. The additional `| any[]` isn't adding anything here. Same for the `value` property below.

Reply:
I knew that it didn't provide any additional checking when I put it in, but I thought that it might make it clear for the human reader that passing an array will behave in a totally different way from anything else.

Reply:
I know what you mean, but in reality unfortunately this is the experience you get in an editor where these typings are used. `any | any[]` is indistinguishable from `any`. 

![image](https://user-images.githubusercontent.com/16928427/29895822-69c6996c-8d8f-11e7-9b47-86382c534d28.png)

As an aside, why didn't `string | number | string[] | number[]` work? Maybe we don't need `any` after all. In the test you linked I see `string | number | (string | number)[]` mentioned which is slightly different.

Reply:
Ok, I understand. Fixed.

-------

I get the following error messages I get when I use `string | number | string[] | number[]`:

```
lib/ToggleButtonGroup.d.ts(7,11): error TS2430: Interface 'ToggleButtonGroupProps' incorrectly extends interface 'HTMLProps<ToggleButtonGroup>'.
  Types of property 'defaultValue' are incompatible.
    Type 'string | number | (string | number)[]' is not assignable to type 'string | string[]'.
      Type 'number' is not assignable to type 'string | string[]'.
```

and

```
lib/ToggleButtonGroup.d.ts(7,11): error TS2430: Interface 'ToggleButtonGroupProps' incorrectly extends interface 'HTMLProps<ToggleButtonGroup>'.
  Types of property 'value' are incompatible.
    Type 'string | number | string[] | number[]' is not assignable to type 'string | number | string[]'.
      Type 'number[]' is not assignable to type 'string | number | string[]'.
        Type 'number[]' is not assignable to type 'string[]'.
          Type 'number' is not assignable to type 'string'.
```